aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -52,16 +53,14 @@ pub enum SessionId {
 
 impl SessionId {
     pub fn txn(txn: &SignatureCheckedTransaction) -> Self {
-        Self::Txn {
-            sender: txn.sender(),
-            sequence_number: txn.sequence_number(),
-        }
+        Self::txn_meta(&TransactionMetadata::new(&txn.clone().into_inner()))
     }
 
     pub fn txn_meta(txn_data: &TransactionMetadata) -> Self {
         Self::Txn {
             sender: txn_data.sender,
             sequence_number: txn_data.sequence_number,
+            script_hash: txn_data.script_hash.clone(),
```

Comment:
should this be called txn_data_hash?

Reply:
Not sure what you mean. It's the script hash, not transaction's data hash (which by name indicates inputs and other transaction metadata as well IMO)